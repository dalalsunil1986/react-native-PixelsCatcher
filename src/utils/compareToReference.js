/* @flow */
import network from './network';

// Throws exception
const compareToReference = async (snapshotName: string, base64: string): Promise<boolean> => {
  const response = await network.postBase64({
    base64,
    fileName: `${snapshotName}.png`,
  });

  if (response.status !== 200) {
    throw new Error(`Invalid status ${response.status}`);
  }
  const responseJSON = await response.json();

  if (responseJSON.result !== 'OK') {
    throw new Error(`Files mistmatch with ${responseJSON.info.differentPixelsCount} pixels`);
  }

  return true;
};

export default compareToReference;
