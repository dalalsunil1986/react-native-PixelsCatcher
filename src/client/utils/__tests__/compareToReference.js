/* @flow */
import network from '../network';
import compareToReference from '../compareToReference';

jest.mock('../network', () => ({ postBase64: jest.fn() }));

describe('compareToReference', () => {
  it('Returns failure if HTTP status is not 200', async () => {
    const snapshotName = 'snapshotName';
    const base64 = 'base64 data';

    (network.postBase64: any).mockImplementationOnce(() => ({ status: 404 }));

    const failure = await compareToReference(snapshotName, base64);
    expect(failure).toMatchSnapshot();
  });

  it('Returns failure if result is not OK', async () => {
    const snapshotName = 'snapshotName';
    const base64 = 'base64 data';

    (network.postBase64: any).mockImplementationOnce(() => ({
      status: 200,
      json: async () => ({
        result: 'ERROR',
        info: { differentPixelsCount: 1 },
      }),
    }));

    const failure = await compareToReference(snapshotName, base64);
    expect(failure).toMatchSnapshot();
  });

  it('Returns nothing if image matches the reference', async () => {
    const snapshotName = 'snapshotName';
    const base64 = 'base64 data';

    (network.postBase64: any).mockImplementationOnce(() => ({
      status: 200,
      json: async () => ({
        result: 'OK',
        info: { differentPixelsCount: 0 },
      }),
    }));

    const result = await compareToReference(snapshotName, base64);
    expect(result).toBe(undefined);
  });
});
