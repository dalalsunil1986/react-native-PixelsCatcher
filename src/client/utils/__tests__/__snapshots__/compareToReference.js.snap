// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compareToReference Returns failure if HTTP status is not 200 1`] = `"Invalid status 404"`;

exports[`compareToReference Returns failure if result is not OK 1`] = `"Files mismatch with 1 pixels"`;
